TESTING.md

```diff
@@ -1,89 +1,483 @@
-- [ ] Tests need to be as minimalistic as possible
-- [ ] Tests should run properly for future versions without major changes
-- [ ] Avoid hardcoded values
-- [ ] Use smart and uniform locator strategies
-- [ ] Prefer fluid waits over hardcoded waits
-- [ ] Favor URL’s for navigation over UI interaction
-- [ ] Include only necessary files
-- [ ] Test code needs to be [maintainable](https://testautomationpatterns.org/wiki/index.php/MAINTAINABLE_TESTWARE)
-- [ ] Test names should be descriptive
-- [ ] Test data should be generated dynamically
+## Testing strategy
 
-### Cypress
+### Defining the scope
 
-- [ ] Test file name has the following format: Helm chart name + spec (ex: `wordpress_spec.js`)
-- [ ] No `describe()` blocks for Cypress test
-- [ ] Aim to have an assertion after every command to avoid flakiness, taking advantage of Cypress retry-ability
-- [ ] Test description is a sentence with the following format: Expected result summary, starting with a verb, in third person, no dots at the end of the sentence (ex: `it('checks if admin can edit a site', ()`)
-- [ ] Respect the folder structure recommended by Cypress:
+The general aim of the tests should be to verify the Chart package works as expected. As such, the focus IS NOT on the application OR the container images, which should be regarded as trustful components (i.e. they should have been respectively tested at a previous stage), but in the Chart itself and the different features it provides. It is expected though to assert the CORE functionality (or functionalities) of the application works, but checks defined in this repository should never aim to replace the official test suite.
+
+Some examples on the suitability of tests for the `bitnami/wordpress` chart:
+ * ✅ Creating a blog post (represents the CORE functionality of the asset)
+ * ❌ Creating a comment in a post (far too specific, not useful)
+ * ❌ Installing a plugin through the admin panel (far too specific, not useful)
+ * ✅ Specifying a different UID using the `containerSecurityContext.runAsUser` in `values.yaml` and checking it (checks a feature intrinsic to the Chart)
+
+The tests may be regarded as _deployment_ tests since their goal is to verify that the software is correctly deployed with all the inherent features. Both functional and non-functional characteristics are evaluated in these tests, focusing on the installation aspect.
+
+### The strategy
+
+Before writing any test scenario, understand the primary purpose of the chart and its components. Take a look at [the documentation about the chart under test](https://github.com/bitnami/charts/tree/main/bitnami), explore the different templates and configurations in `values.yaml` and glance over the [docker image documentation](https://github.com/bitnami?q=docker&type=all&language=&sort=). This will give you a solid base for creating valuable test scenarios.
```

Comment:
```suggestion
Before writing any test scenario, understand the primary purpose of the chart and its components. Take a look at [the documentation about the chart under test](https://github.com/bitnami/charts/tree/main/bitnami), explore the different templates and configurations in `values.yaml` and glance over the [docker image documentation](https://github.com/bitnami/containers). This will give you a solid base for creating valuable test scenarios.
```